### DISCUSSION TITLE
In HA mode, the node assigns itself the IP address of an inaccessible node

### DISCUSSION THREAD

**Discussion by zeus** [rancher/rke2#4319] [Q&A]
**Environmental Info:**
RKE2 Version: 
In HA mode, the node assigns itself the address of an inaccessible node

Node(s) CPU architecture, OS, and Version: 
Linux alpha 5.15.0-56-generic #62-Ubuntu SMP Tue Nov 22 19:54:14 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
```
NAME    STATUS   ROLES                       AGE    VERSION          INTERNAL-IP     EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
alpha   Ready    control-plane,etcd,master   169d   v1.24.8+rke2r1   185.**.**.21   <none>        Ubuntu 22.04.1 LTS   5.15.0-56-generic   containerd://1.6.8-k3s1
beta    Ready    control-plane,etcd,master   169d   v1.24.8+rke2r1   185.**.**.23   <none>        Ubuntu 22.04.1 LTS   5.15.0-56-generic   containerd://1.6.8-k3s1
gamma   Ready    control-plane,etcd,master   169d   v1.24.8+rke2r1   185.**.**.26   <none>        Ubuntu 22.04.1 LTS   5.15.0-43-generic   containerd://1.6.8-k3s1

```

**Describe the bug:**
The cluster contains 3 physical servers in one datacenter each with Internet-accessible IP addresses

In the absence of availability of one of the machines for a short period of time, the second assigns itself its IP address.

As a result, there is a situation with a doubling of IP addresses and one of the machines becomes inaccessible from the Internet. After reconnecting the network on it, the IP address of the machine that "took the IP" is removed and the cluster resumes normal operation.

This is unexpected behavior and i can't find any docs about it

**Comment by brandond** [comment]
> In the absence of availability of one of the machines for a short period of time, the second assigns itself its IP address.

I have no idea what would cause this, nor have I seen anyone mention this in the past. Do you perhaps have something else deployed to the nodes that's managing addresses or virtual IPs?

**Reply by brandond** [reply]
Can you provide rke2-server logs and/or `kubectl get node -o yaml` output when the machine with the "doubled IP" is present in the cluster?
